### PR TITLE
Settings Navigation [CV2-3294]

### PR DESCRIPTION
**Pull request by brianfleming**
## Description

Follows the great work by @danielevalverde , I tried my hand at creating a new side menu navigation for the settings area per the ticket [CV2-3294](https://meedan.atlassian.net/browse/CV2-3294)

References: [CV2-3294](https://meedan.atlassian.net/browse/CV2-3294)

## Type of change

- [x] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?
Manul testing and updates to unit tests

### AFTER
Screenshot 2024-01-24 at 1 57 04 PM

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [x] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)

[CV2-3294]: https://meedan.atlassian.net/browse/CV2-3294?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CV2-3294]: https://meedan.atlassian.net/browse/CV2-3294?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/components/team/TeamDetails.js
Comment (brianfleming) [inline src/app/components/team/TeamDetails.js:1]: These changes I just noticed randomly, they are attributes that are not necessary for our CDS TextField

File: src/app/components/drawer/Projects/SettingsComponent.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+  },
+});
+
+const SettingsComponent = ({
+  team,
+  params,
+  intl,
+}) => {
+  const { tab } = params;
+  const userRole = UserUtil.myRole(window.Check.store.getState().app.context.currentUser, team.slug);
```
Comment (brianfleming) [inline src/app/components/drawer/Projects/SettingsComponent.js:95]: I got this from https://github.com/meedan/check-web/blob/da765780169f0bc1e4357f84a0c2bcf3e98f9345/src/app/components/Sandbox.js#L335 as this seemed the way to get this information in a function component but I am unsure how to fix the unit test in the PR

**Comment by brianfleming** [comment]
Not really ready for review. Just wanted to see if someone can give me a bit of guidance on updating the test in this PR in order to work with the new file. cc @caiosba @dariusk @danielevalverde

**Comment by caiosba** [comment]
> Not really ready for review. Just wanted to see if someone can give me a bit of guidance on updating the test in this PR in order to work with the new file. cc @caiosba @dariusk @danielevalverde

@brianfleming, we can mock that value, I pushed a fix for it: https://github.com/meedan/check-web/pull/1810/commits/41e730d83f84f345e8c078f7539cbdcdd8c6ab6f - happy to explain further.

**Comment by brianfleming** [comment]
> 20ad4b0

oh, I see what you did there. Thanks for the assist @caiosba ! Much appreciated. Going to push up a integration test run to make sure its good to go.

**Comment by brianfleming** [comment]
Ok, I have messed with this enough. I don't understand why the last 3 integration test failures happen and could use some reviewer assistance again. I'm clearly missing something, but I'm just not seeing it. Any help would be appreciated

**Comment by caiosba** [comment]
@danielevalverde, as per our conversation, please take a first look into those integration tests... please assign it for my review if you need help.

**Comment by brianfleming** [comment]
> @danielevalverde, as per our conversation, please take a first look into those integration tests... please assign it for my review if you need help.

@danielevalverde is my hero!

**Comment by brianfleming** [comment]
One thing I noticed that might help diagnose the issues on this branch. Occasionally adding a new language works without needing to refresh the page to see the new language, but adding another one does not succeed at all afterwards

**Comment by caiosba** [comment]
@brianfleming , I gave an idea to @danielevalverde to just ask for the GraphQL fields needed for the new page - let's see what she finds.

**Comment by brianfleming** [comment]
@danielevalverde just wanted to check in on this PR so it doesn't get too stale. Did you get a chance to try the graphql idea?

File: src/app/components/team/Statuses/StatusesComponent.js [hunk 2]
```diff
@@ -93,7 +93,9 @@ const StatusesComponent = ({ team, setFlashMessage }) => {
 
   const handleDelete = ({ status_id, fallback_status_id }) => {
     const onCompleted = () => {
+      setStatuses(statuses.filter(s => s.id !== status_id));
```
Comment (brianfleming) [inline src/app/components/team/Statuses/StatusesComponent.js:96]: @danielevalverde I believe adding this resolves the delete not happening on the screen right away. Looking at the changes Darius made to the Language area, specifically `LanguageListItem.js line 133` in this PR, he is filtering out of the array the language that was deleted, so I just did the same, which seems to resolve that issue

File: src/app/components/team/Statuses/EditStatusDialog.js [hunk 3]
```diff
@@ -26,11 +24,18 @@ const EditStatusDialog = ({
 }) => {
   const [statusLabel, setStatusLabel] = React.useState(status ? status.label : '');
   const [statusDescription, setStatusDescription] = React.useState(status ? status.locales[defaultLanguage].description : '');
-  const [statusColor, setStatusColor] = React.useState(status ? status.style.color : 'var(--textPrimary)');
+  const [statusColor, setStatusColor] = React.useState(status ? status.style.color : '#567bff');
```
Comment (brianfleming) [inline src/app/components/team/Statuses/EditStatusDialog.js:27]: for reviewers, I noticed that we were using a css var before, but the color picker doesn't correctly set the color unless you use a hex code, just an FYI for this change

File: src/app/components/team/Statuses/StatusesComponent.js (hunk 2 above)
Comment (danielevalverde) [reply src/app/components/team/Statuses/StatusesComponent.js:96]: Yaaay!! I tested locally and its working for me :heart_eyes: :heart_eyes:

File: src/app/components/team/Statuses/EditStatusDialog.js [hunk 4]
```diff
@@ -26,11 +24,18 @@ const EditStatusDialog = ({
...
   const [statusMessage, setStatusMessage] = React.useState(status ? status.locales[defaultLanguage].message : '');
   const [statusMessageEnabled, setStatusMessageEnabled] = React.useState(status ? Boolean(status.should_send_message) : false);
   const [showSaveConfirmDialog, setShowSaveConfirmDialog] = React.useState(false);
 
+  const handleCancel = () => {
```
[the remark this answers is not in the document]
Comment (danielevalverde) [reply src/app/components/team/Statuses/EditStatusDialog.js:32]: @brianfleming , It does fix the case where when creating a new status and subsequently attempting to add another one, the field retains the data from the previously created state.
However, we still have an issue here
- After creating a new status and then attempting to edit it, the dialog appears empty, failing to retain the data of the status that was clicked for editing :grimacing:

[the remark this answers is not in the document]
Comment (brianfleming) [reply src/app/components/team/Statuses/EditStatusDialog.js:32]: hmmm...these statuses are a real pain. I can look at that and see if I can resolve anything with your bullet point.
there seems to be 1 integration test failing now, the `should customize status` which I think i might need your help with

[the remark this answers is not in the document]
Comment (brianfleming) [reply src/app/components/team/Statuses/EditStatusDialog.js:32]: Ok @danielevalverde I think my last push fixes your new bullet point, for some reason we weren't handling the status label the same way as other text fields

**Comment by brianfleming** [comment]
Ok, things are looking better @danielevalverde only integration test issues (of course)

**Comment by danielevalverde** [comment]
> Ok, things are looking better @danielevalverde only integration test issues (of course)

@brianfleming, just sent a commit with the test fix :)

**Comment by brianfleming** [comment]
Woot! @danielevalverde green checks! Thank you so much for your help!

dabest

@caiosba can you take another look since this branch has been pretty long lived at this point?

**Comment by caiosba** [comment]
Yay! Great work, team! I'm going to review 👀

**Review by caiosba — APPROVED** [review]
I tested all settings pages locally. All good. When we merge this, I suggest we ask the QA team to do a more thoroughly test on QA.

**Pull request merged**